test/mpi/include/mtest_common.h

```diff
@@ -33,6 +33,49 @@ long MTestArgListGetLong_with_default(MTestArgList * head, const char *arg, long
 mtest_mem_type_e MTestArgListGetMemType(MTestArgList * head, const char *arg);
 void MTestArgListDestroy(MTestArgList * head);
 
+#define MTEST_DTP_DECLARE(name) \
```

Comment:
This looks like magic. Could we wrap these declarations up in a struct to be passed into the other APIs?

Reply:
That sounds like a good idea. Let me try...

Reply:
Added the last commit to refactor into `struct` and `inline functions`.